modules/mbedtls/Kconfig.crypto

```diff
@@ -371,12 +339,6 @@ config MBEDTLS_PSA_CRYPTO_CLIENT
 	depends on BUILD_WITH_TFM || MBEDTLS_PSA_CRYPTO_C
```

Comment:
Actually, how about we remove this Kconfig option and only use `PSA_CRYPTO_CLIENT`? They're sort of duplicates and `CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT` already doesn't map to a single `#define` in the Mbed TLS configuration file.